pkg/smi/health.go

```diff
@@ -0,0 +1,62 @@
+package smi
+
+import (
+	smiAccess "github.com/servicemeshinterface/smi-sdk-go/pkg/apis/access/v1alpha3"
+	smiSpecs "github.com/servicemeshinterface/smi-sdk-go/pkg/apis/specs/v1alpha4"
+	smiSplit "github.com/servicemeshinterface/smi-sdk-go/pkg/apis/split/v1alpha2"
+	extensionsClientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+
+	"github.com/openservicemesh/osm/pkg/version"
+)
+
+// HealthChecker has SMI kube config to access SMI CRDS
+type HealthChecker struct {
+	SMIClient *extensionsClientset.Clientset
```

Comment:
What you have is very close to what I meant, but this would be the last step needed so then you can instantiate a `HealthChecker` with either a `*extensionsClientset.Clientset` (like you do in `main()`) or the [fake](https://pkg.go.dev/k8s.io/apiextensions-apiserver@v0.21.0/pkg/client/clientset/clientset/fake#Clientset) (for unit tests):
```suggestion
	SMIClient *extensionsClientset.Interface
```